app/api/phones/route.ts

```diff
@@ -0,0 +1,83 @@
+import { NextRequest } from "next/server";
+import { authMiddleware } from "@/middleware/authMiddleware";
+import { upsertPhone } from "@/lib/supabase/in_process_artist_phones/upsertPhone";
+import { sendSmsVerification } from "@/lib/phones/sendSmsVerification";
+import { validatePhoneNumber } from "@/lib/phones/validatePhoneNumber";
+import { selectArtist } from "@/lib/supabase/in_process_artists/selectArtist";
+import getCorsHeader from "@/lib/getCorsHeader";
+import truncateAddress from "@/lib/truncateAddress";
+
+const corsHeaders = getCorsHeader();
+
+export async function POST(req: NextRequest) {
+  try {
+    const authResult = await authMiddleware(req, { corsHeaders });
+    if (authResult instanceof Response) {
+      return authResult;
+    }
+    const { artistAddress } = authResult;
+
+    // Get phone number from request body
+    const body = await req.json();
+    const { phone_number } = body;
+
+    if (!phone_number || typeof phone_number !== "string") {
+      return Response.json(
+        { message: "phone_number is required and must be a string" },
+        { status: 400, headers: corsHeaders }
+      );
+    }
+
+    // Trim whitespace and validate E.164 format
+    let trimmedPhoneNumber: string;
+    try {
+      trimmedPhoneNumber = validatePhoneNumber(phone_number);
+    } catch (error) {
+      return Response.json(
+        { message: error instanceof Error ? error.message : "Invalid phone number format" },
+        { status: 400, headers: corsHeaders }
+      );
+    }
```

Comment:
Single Responsibility Principle
- actual: body validation is the API endpoint file.
- required: new `lib/phones/validatePhoneRequest.ts`
- reference authMiddleware for implementation details
- use zod for body parsing